testsuite/src/it/java/io/quarkus/ts/startstop/CodeQuarkusTest.java

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * @author Michal Karm Babacek <karm@redhat.com>
+ */
+@Tag("codequarkus")
+public class CodeQuarkusTest {
+
+    private static final Logger LOGGER = Logger.getLogger(CodeQuarkusTest.class.getName());
+
+    public static final String GEN_BASE_DIR = getArtifactGeneBaseDir();
+
+    public static final List<List<CodeQuarkusExtensions>> supportedEx = CodeQuarkusExtensions.partition(4, CodeQuarkusExtensions.Flag.SUPPORTED);
+    public static final List<List<CodeQuarkusExtensions>> notSupportedEx = CodeQuarkusExtensions.partition(1, CodeQuarkusExtensions.Flag.NOT_SUPPORTED);
+    public static final List<List<CodeQuarkusExtensions>> mixedEx = CodeQuarkusExtensions.partition(1, CodeQuarkusExtensions.Flag.MIXED);
+
+    public void testRuntime(TestInfo testInfo, List<CodeQuarkusExtensions> extensions) throws Exception {
+        Process pA = null;
+        File unzipLog = null;
+        File runLogA = null;
+        String cn = testInfo.getTestClass().get().getCanonicalName();
+        String mn = testInfo.getTestMethod().get().getName();
+        LOGGER.info(mn + ": Testing Code Quarkus generator with these " + extensions.size() + " extensions: " + extensions.toString());
+        File appDir = new File(GEN_BASE_DIR + File.separator + "code-with-quarkus");
+        String logsDir = GEN_BASE_DIR + File.separator + "code-with-quarkus-logs";
+        List<String> devCmd = getBuildCommand(MvnCmds.MVNW_DEV.mvnCmds[0]);
+        URLContent skeletonApp = Apps.GENERATED_SKELETON.urlContent;
+        String zipFile = GEN_BASE_DIR + File.separator + "code-with-quarkus.zip";
+
+        try {
+            cleanDir(appDir.getAbsolutePath(), logsDir);
+            Files.createDirectories(Paths.get(logsDir));
+            LOGGER.info("Downloading...");
+            download(extensions, zipFile);
+            LOGGER.info("Unzipping...");
+            unzipLog = unzip(zipFile, GEN_BASE_DIR);
+            LOGGER.info("Running...");
+            runLogA = new File(logsDir + File.separator + "dev-run.log");
+            pA = runCommand(devCmd, appDir, runLogA);
+            // It takes time to download the Internet
+            long timeoutS = 30 * 60;
+            LOGGER.info("Waiting for the web content...");
+            WebpageTester.testWeb(skeletonApp.urlContent[0][0], timeoutS, skeletonApp.urlContent[0][1], false);
+            LOGGER.info("Terminating and scanning logs...");
+            pA.getInputStream().available();
+            processStopper(pA, false);
+            LOGGER.info("Gonna wait for ports closed...");
+            assertTrue(waitForTcpClosed("localhost", parsePort(skeletonApp.urlContent[0][0]), 60),
+                    "Main port is still open.");
+            checkLog(cn, mn, Apps.GENERATED_SKELETON, MvnCmds.MVNW_DEV, runLogA);
+        } finally {
+            if (pA != null) {
+                processStopper(pA, true);
+            }
+            archiveLog(cn, mn, unzipLog);
```

Comment:
I executed `mvn test -Dtest=CodeQuarkusTest -pl testsuite` twice and received error here for the second run (didn't have `clean` step there). The target file already exists from the previous run. I think it should be overwritten instead of throwing an error.

```
[ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0, Time elapsed: 56.798 s <<< FAILURE! - in io.quarkus.ts.startstop.CodeQuarkusTest
[ERROR] mixExtensions{TestInfo}  Time elapsed: 9.89 s  <<< ERROR!
java.nio.file.FileAlreadyExistsException: /Users/rsvoboda/Downloads/ffff/quarkus-startstop/testsuite/target/archived-logs/io.quarkus.ts.startstop.CodeQuarkusTest/mixExtensions/code-with-quarkus.zip.log
	at io.quarkus.ts.startstop.CodeQuarkusTest.testRuntime(CodeQuarkusTest.java:102)
	at io.quarkus.ts.startstop.CodeQuarkusTest.mixExtensions(CodeQuarkusTest.java:135)
...
```